libcfnet/protocol.c

```diff
@@ -88,3 +89,123 @@ Seq *ProtocolOpenDir(AgentConnection *conn, const char *path)
 
     return seq;
 }
+
+bool ProtocolGet(AgentConnection *conn, const char *remote_path,
+                 const char *local_path, const uint32_t file_size, int perms)
+{
+    assert(conn != NULL);
+    assert(remote_path != NULL);
+    assert(local_path != NULL);
+    assert(file_size != 0);
+
+    perms = (perms == 0) ? CF_PERMS_DEFAULT : perms;
+
+    unlink(local_path);
+    FILE *file_ptr = safe_fopen_create_perms(local_path, "wx", perms);
+    if (file_ptr == NULL)
+    {
+        Log(LOG_LEVEL_ERR, "%s: Failed to open file %s (fopen: %s)",
+            __func__, local_path, GetErrorStr());
+        return false;
+    }
+
+    char buf[CF_MSGSIZE] = {0};
+    int to_send = snprintf(buf, CF_MSGSIZE, "GET %d %s",
+                           CF_MSGSIZE, remote_path);
+
+    int ret = SendTransaction(conn->conn_info, buf, to_send, CF_DONE);
+    if (ret == -1)
+    {
+        Log(LOG_LEVEL_ERR,
+            "%s: Failed to send request for file %s on host %s",
+            __func__, remote_path, conn->this_server);
```

Comment:
We want our log messages (verbose and above) to be user oriented. The function name doesn't belong here.